awkward-cpp/CMakeLists.txt

```diff
@@ -54,31 +55,24 @@ target_include_directories(awkward-parent INTERFACE rapidjson/include)
 add_subdirectory(header-only EXCLUDE_FROM_ALL)
 target_link_libraries(awkward-parent INTERFACE awkward::growable-buffer)
 
-# First tier: cpu-kernels (object files, static library, and dynamic library).
-add_library(awkward-cpu-kernels-objects OBJECT ${CPU_KERNEL_SOURCES})
-set_target_properties(awkward-cpu-kernels-objects PROPERTIES POSITION_INDEPENDENT_CODE ON)
-target_link_libraries(awkward-cpu-kernels-objects PUBLIC awkward-parent)
+# First tier: cpu-kernels
+add_library(awkward-cpu-kernels SHARED ${CPU_KERNEL_SOURCES})
+target_link_libraries(awkward-cpu-kernels PUBLIC awkward-parent)
 set_target_properties(
-  awkward-cpu-kernels-objects
+  awkward-cpu-kernels
   PROPERTIES CXX_VISIBILITY_PRESET hidden
              VISIBILITY_INLINES_HIDDEN ON
              CXX_EXTENSIONS NO)
 
-add_library(awkward-cpu-kernels-static STATIC $<TARGET_OBJECTS:awkward-cpu-kernels-objects>)
-set_property(TARGET awkward-cpu-kernels-static PROPERTY POSITION_INDEPENDENT_CODE ON)
-target_link_libraries(awkward-cpu-kernels-static PUBLIC awkward-parent)
-
-add_library(awkward-cpu-kernels SHARED $<TARGET_OBJECTS:awkward-cpu-kernels-objects>)
-target_link_libraries(awkward-cpu-kernels PUBLIC awkward-parent)
+# Second tier: python
+find_package(Python REQUIRED COMPONENTS Interpreter Development)
+find_package(pybind11 CONFIG REQUIRED)
 
-# Second tier: libawkward (object files, static library, and dynamic library).
-add_library(awkward-objects OBJECT ${LIBAWKWARD_SOURCES})
-set_target_properties(awkward-objects PROPERTIES POSITION_INDEPENDENT_CODE 1)
-target_compile_definitions(awkward-objects PRIVATE LIBAWKWARD_EXPORT_SYMBOL=EXPORT_SYMBOL)
```

Comment:
I assume that we don't need to optionally export these symbols any more; we are building two libraries with separate source trees now.